crates/agent_ui/src/acp/model_selector.rs

```diff
@@ -192,8 +192,16 @@ impl PickerDelegate for AcpModelPickerDelegate {
         }
     }
 
-    fn dismissed(&mut self, _: &mut Window, cx: &mut Context<Picker<Self>>) {
+    fn dismissed(&mut self, window: &mut Window, cx: &mut Context<Picker<Self>>) {
         cx.emit(DismissEvent);
+        cx.spawn_in(window, async move |handle, cx| {
+            handle
+                .update_in(cx, |picker, window, cx| {
+                    picker.set_query("", window, cx);
+                })
+                .ok();
+        })
+        .detach();
```

Comment:
Should we also be doing this in the `confirm` method above?

Also, can we do something a little lighter weight here? since we don't need the async


```suggestion
        cx.defer_in(window, |picker, window, cx| {
            picker.set_query("", window, cx);
        });
```

Reply:
Nice, this code for cleaning up make more sense. TIL didn't know defer_in existed.

For adding this in confirmed: Actually we are raising DismissEvent from it which in turns calls dismissed method.

```
confirmed sends `DismissEvent` → triggers dismissal → triggers `menu::Cancel` action → calls `cancel()` → calls `dismissed()` → clears query.
```

I also one more thing I realised during my testing was that we don't need to raise emit DismissEvent from dismissed it wouldn't do anything as the picker was already dismissed. So I have removed that as well.

ref:

```
pub fn cancel(&mut self, _: &menu::Cancel, window: &mut Window, cx: &mut Context<Self>) {
    if self.delegate.should_dismiss() {
        self.delegate.dismissed(window, cx);  
        cx.emit(DismissEvent);                
    }
}

```